fs/copy.ts

```diff
@@ -34,6 +33,12 @@ interface InternalCopyOptions extends CopyOptions {
   isFolder?: boolean;
 }
 
+function assertIsDate(date: Date | null, name: string): asserts date is Date {
+  if (date === null) {
+    throw new ReferenceError(`${name} is unavailable`);
```

Comment:
This usage of `ReferenceError` seems strange to me (It's not about variable being unavailable). I think this should be `TypeError` or `Error`.

Reply:
I was going to use `TypeError` originally. But we pass [`Deno.FileInfo.atime`](https://deno.land/api?s=Deno.FileInfo#prop_atime) and [`Deno.FileInfo.mtime`](https://deno.land/api?s=Deno.FileInfo#prop_mtime) to this function, which can only be `null` on some platforms (see documentation). So I went with `ReferenceError`. WDYT?

Reply:
`ReferenceError` is thrown for the case where undeclared 'variable' used like the below:

```ts
const foo = a; // a is not declared
```

There seems no undeclared variable involved in this case. So I think this case doesn't fit to `ReferenceError`

Reply:
Ah, yeah, fair point. I'll go with `Error`.

Reply:
I went with `Error`.